> `> Task :Decompiler:compileSleighMac_arm_64ExecutableSleighCpp FAILED`

Well that's fun. It's trying to build the arm64 version of the binaries, and failing when importing headers, like:

```
In file included from /private/tmp/nix-build-ghidra-10.2.3.drv-0/source/Ghidra/Features/Decompiler/src/decompile/cpp/types.h:22:
In file included from /nix/store/8p1kax7r4mljm5zc7q2z775qp0pp82c4-libcxx-11.1.0-dev/include/c++/v1/cstdint:144:
In file included from /nix/store/8p1kax7r4mljm5zc7q2z775qp0pp82c4-libcxx-11.1.0-dev/include/c++/v1/stdint.h:123:
In file included from /nix/store/ifsryrawz48d3g4am68kcm8mf0xy6i5j-clang-11.1.0-lib/lib/clang/11.1.0/include/stdint.h:52:
In file included from /nix/store/xma9cryla3v8gmsmjjgcsh6wy3nd2r7b-Libsystem-1238.60.2/include/stdint.h:52:
In file included from /nix/store/xma9cryla3v8gmsmjjgcsh6wy3nd2r7b-Libsystem-1238.60.2/include/sys/_types.h:32:
/nix/store/xma9cryla3v8gmsmjjgcsh6wy3nd2r7b-Libsystem-1238.60.2/include/sys/cdefs.h:761:2: error: Unsupported architecture
#error Unsupported architecture
```

or

```
In file included from /private/tmp/nix-build-ghidra-10.2.3.drv-0/source/Ghidra/Features/Decompiler/src/decompile/cpp/types.h:22:
In file included from /nix/store/8p1kax7r4mljm5zc7q2z775qp0pp82c4-libcxx-11.1.0-dev/include/c++/v1/cstdint:144:
In file included from /nix/store/8p1kax7r4mljm5zc7q2z775qp0pp82c4-libcxx-11.1.0-dev/include/c++/v1/stdint.h:123:
In file included from /nix/store/ifsryrawz48d3g4am68kcm8mf0xy6i5j-clang-11.1.0-lib/lib/clang/11.1.0/include/stdint.h:52:
In file included from /nix/store/xma9cryla3v8gmsmjjgcsh6wy3nd2r7b-Libsystem-1238.60.2/include/stdint.h:52:
/nix/store/xma9cryla3v8gmsmjjgcsh6wy3nd2r7b-Libsystem-1238.60.2/include/sys/_types.h:55:9: error: unknown type name '__int64_t'
typedef __int64_t       __darwin_blkcnt_t;      /* total blocks */
        ^
```

Maybe I need a newer version of the SDK?

--- a/pkgs/tools/security/ghidra/build.nix
+++ b/pkgs/tools/security/ghidra/build.nix
@@ -19,13 +19,13 @@
 let
   pkg_path = "$out/lib/ghidra";
   pname = "ghidra";
-  version = "10.2.2";
+  version = "10.2.3";
 
   src = fetchFromGitHub {
     owner = "NationalSecurityAgency";
     repo = "Ghidra";
     rev = "Ghidra_${version}_build";
-    sha256 = "sha256-AiyY6mGM+jHu9n39t/cYj+I5CE+a3vA4P0THNEFoZrk=";
+    sha256 = "sha256-YhjKRlFlF89H05NsTS69SB108rNiiWijvZZY9fR+Ebc=";
   };
 
   desktopItem = makeDesktopItem {

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -4784,7 +4784,8 @@ with pkgs;
 
   ghdorker = callPackage ../tools/security/ghdorker { };
 
-  ghidra = callPackage ../tools/security/ghidra/build.nix { };
+  ghidra = if stdenv.isDarwin then darwin.apple_sdk_11_0.callPackage ../tools/security/ghidra/build.nix {}
+  else callPackage ../tools/security/ghidra/build.nix {};
 
   ghidra-bin = callPackage ../tools/security/ghidra { };
 

